### PR TITLE
chore(*): upgrade kubectl to version with ARM support

### PR DESCRIPTION
**Pull request by lukidzi**
### Summary

upgrade kubectl to version with ARM support

### Full changelog

* upgrade kubectl to version with ARM support

### Issues resolved

Part of #237 
### Documentation

- [ ] Link to the website [documentation PR](https://github.com/kumahq/kuma-website/pull/XXX)

### Testing

- [ ] Unit tests
- [ ] E2E tests
- [ ] Manual testing on Universal
- [ ] Manual testing on Kubernetes

### Backwards compatibility

- [ ] Update [`UPGRADE.md`](/UPGRADE.md) with any steps users will need to take when upgrading.
- [ ] Add `backport-to-stable` label if the code follows our [backporting policy](/CONTRIBUTING.md#backporting)

### CODE REVIEW

File: mk/dev.mk [hunk 1]
```diff
@@ -13,7 +13,7 @@ SHELLCHECK_VERSION := v0.8.0
 YQ_VERSION := v4.24.2
 
 CI_KUBEBUILDER_VERSION ?= 2.3.2
-CI_KUBECTL_VERSION ?= v1.18.14
+CI_KUBECTL_VERSION ?= v1.23.5
```
Comment (michaelbeaumont) [inline mk/dev.mk:16]: Does this affect the minimum k8s version we support?

Comment (jakubdyszkiewicz) [reply mk/dev.mk:16]: That's just a client, right? We have E2E for old K8S.

Comment (michaelbeaumont) [reply mk/dev.mk:16]: Good point

**Review by jakubdyszkiewicz — APPROVED** [review]

File: mk/dev.mk (hunk 1 above)
Comment (lukidzi) [reply mk/dev.mk:16]: Yes, I think it's connected with this: https://kubernetes.io/releases/version-skew-policy/#kubectl also the policy changes and older versions might not support new versions of `kube-apiserver`. The tool version is (AFAIK) only used for unit tests.

**Pull request merged**